src/platforms/esp32/main/sys.c

```diff
@@ -117,13 +136,14 @@ static void receive_events(GlobalContext *glb, TickType_t wait_ticks)
             return;
         }
 
-        do {
```

Comment:
I would like to explain this change:  Without any change, the application can crash here, if the event handler removes and deleted the listener.  So some care needs to be taken to find the end of the list (especially if the EventDescriptor being deleted is the last item in the list!)

The obvious deficiency in this change is that there are two iterations of the list — one to find the length, and the other to (potentially) call the handler (O(2n)).  However, the number of special cases to check seemed excessive, so I opted for the more semantically clear iteration over n-many items, which is guaranteed to succeed, even if the first, last, or a middle item is removed from the list.